core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -172,7 +172,7 @@ void *impl_allocate_common(const Cuda &exec_space, const char *arg_label,
 #error CUDART_VERSION undefined!
 #elif (defined(KOKKOS_ENABLE_IMPL_CUDA_MALLOC_ASYNC) && CUDART_VERSION >= 11020)
   cudaError_t error_code;
-  if (arg_alloc_size >= memory_threshold_g) {
+  if (arg_alloc_size >= (size_t)memory_threshold_g) {
```

Comment:
Considering that this and L350 below are the only use of `memory_threshold_g`, please change its definition from `int` to `size_t` instead of casting.